src/protocol/boolean/bitwise_sum.rs

```diff
@@ -149,50 +150,112 @@ mod tests {
     }
 
     #[tokio::test]
-    pub async fn fp31_basic() -> Result<(), BoxError> {
+    pub async fn fp31_basic() -> Result<(), Error> {
         let c = Fp31::from;
 
-        assert_eq!(c(0_u8), bits_to_field(&bitwise_sum(c(0), c(0)).await?));
-        assert_eq!(c(1), bits_to_field(&bitwise_sum(c(0), c(1)).await?));
-        assert_eq!(c(1), bits_to_field(&bitwise_sum(c(1), c(0)).await?));
-        assert_eq!(c(2), bits_to_field(&bitwise_sum(c(1), c(1)).await?));
+        assert_eq!(0, bits_to_value(&bitwise_sum(c(0_u32), c(0)).await?));
+        assert_eq!(1, bits_to_value(&bitwise_sum(c(0), c(1)).await?));
+        assert_eq!(1, bits_to_value(&bitwise_sum(c(1), c(0)).await?));
+        assert_eq!(2, bits_to_value(&bitwise_sum(c(1), c(1)).await?));
+        assert_eq!(32, bits_to_value(&bitwise_sum(c(16), c(16)).await?));
+        assert_eq!(60, bits_to_value(&bitwise_sum(c(30), c(30)).await?));
 
         Ok(())
     }
 
     #[tokio::test]
-    pub async fn fp_32bit_prime_basic() -> Result<(), BoxError> {
+    pub async fn fp_32bit_prime_basic() -> Result<(), Error> {
         let c = Fp32BitPrime::from;
 
-        assert_eq!(c(0_u32), bits_to_field(&bitwise_sum(c(0), c(0)).await?));
-        assert_eq!(c(1), bits_to_field(&bitwise_sum(c(0), c(1)).await?));
-        assert_eq!(c(1), bits_to_field(&bitwise_sum(c(1), c(0)).await?));
-        assert_eq!(c(2), bits_to_field(&bitwise_sum(c(1), c(1)).await?));
+        assert_eq!(0, bits_to_value(&bitwise_sum(c(0_u32), c(0)).await?));
```

Comment:
I realized that `BitwiseSum` outputs a sequence of shared bits that is not necessarily in the inputs' field. An overflow (in the inputs' field) will not be rounded; rather the output will have +1 bit to preserve the potential carry. Remove `F::from` from asserts, and use `bits_to_value` to reconstruct values in integers, not in fields.